pageserver/src/timelines.rs

```diff
@@ -299,30 +309,24 @@ fn bootstrap_timeline(
     Ok(timeline)
 }
 
-pub(crate) fn get_timelines(
+pub(crate) fn get_local_timelines(
     tenant_id: ZTenantId,
     include_non_incremental_logical_size: bool,
-) -> Result<Vec<TimelineInfo>> {
+) -> Result<Vec<(ZTimelineId, LocalTimelineInfo)>> {
     let repo = tenant_mgr::get_repository_for_tenant(tenant_id)
         .with_context(|| format!("Failed to get repo for tenant {}", tenant_id))?;
 
-    Ok(repo
-        .list_timelines()
-        .with_context(|| format!("Failed to list timelines for tenant {}", tenant_id))?
-        .into_iter()
-        .filter_map(|timeline| match timeline {
-            RepositoryTimeline::Local { timeline, id } => Some((id, timeline)),
-            RepositoryTimeline::Remote { .. } => None,
-        })
-        .map(|(timeline_id, timeline)| {
-            TimelineInfo::from_dyn_timeline(
-                tenant_id,
-                timeline_id,
-                timeline.as_ref(),
+    let mut local_timelines = vec![];
```

Comment:
NIT: `Vec::with_capacity()` of the listed timelines would be slightly more reasonable.